File: office_server/src/main/kotlin/foatto/office/report/mUS.java
//package foatto.office.report;
//
//import foatto.core_server.app.server.AliasConfig;
//import foatto.core_server.app.server.UserConfig;
//import foatto.core_server.app.server.column.ColumnBoolean;
//import foatto.core_server.app.server.column.ColumnInt;
//import foatto.core_server.app.server.column.ColumnString;
//import foatto.core_server.app.server.mAbstractReport;
//import foatto.core_server.ds.CoreDataServer;
//import foatto.core_server.ds.CoreDataWorker;
//
//import java.util.HashMap;
//
//public class mUS extends mAbstractReport {
//
//    private ColumnInt columnReportUser = null;
//    private ColumnBoolean columnSumOnly = null;
//
////----------------------------------------------------------------------------------------------------------------------
//
//    public void init( CoreDataServer dataServer, CoreDataWorker dataWorker, AliasConfig aliasConfig, UserConfig userConfig,
//                      HashMap<String,Integer> hmParentData, int id, boolean isOldVersion ) {
//
//        super.init( dataServer, dataWorker, aliasConfig, userConfig, hmParentData, id, isOldVersion );
//
//
////----------------------------------------------------------------------------------------------------------------------
//
//        tableName = "OFFICE_report";
//
////----------------------------------------------------------------------------------------------------------------------
//
//        columnID = new ColumnInt( tableName, "id" );
//
////----------------------------------------------------------------------------------------------------------------------
//
//            ColumnInt columnUserID = new ColumnInt( "SYSTEM_users", "id" );
//        columnReportUser = new ColumnInt( tableName, "user_id", columnUserID, 0 /*userConfig.getUserID()*/ );
//        ColumnString columnUserName = new ColumnString( "SYSTEM_users", "full_name", "По пользователю", STRING_COLUMN_WIDTH );
//            columnUserName.setSelectorAlias( "system_user_people" );
//            columnUserName.addSelectorColumn( columnReportUser, columnUserID );
//            columnUserName.addSelectorColumn( columnUserName );
//
//        columnSumOnly = new ColumnBoolean( tableName, "sum_only", "Выводить только суммы", false );
//
////----------------------------------------------------------------------------------------------------------------------
//
//        alFormHiddenColumn.add( columnID );
//        alFormHiddenColumn.add( columnReportUser );
//
//        alFormColumn.add( columnUserName );
//        alFormColumn.add( columnSumOnly );
//    }
//
//    public ColumnInt getColumnReportUser() { return columnReportUser; }
//    public ColumnBoolean getColumnSumOnly() { return columnSumOnly; }
//}

File: office_server/src/main/kotlin/foatto/office/report/mUS.kt
package foatto.office.report

import foatto.core_server.app.iApplication
import foatto.core_server.app.server.AliasConfig
import foatto.core_server.app.server.UserConfig
import foatto.core_server.app.server.column.ColumnBoolean
import foatto.core_server.app.server.column.ColumnInt
import foatto.core_server.app.server.column.ColumnString
import foatto.core_server.app.server.mAbstractReport
import foatto.sql.CoreAdvancedStatement

class mUS : mAbstractReport() {

    lateinit var columnReportUser: ColumnInt
        private set
    lateinit var columnSumOnly: ColumnBoolean
        private set

    override fun init(
        application: iApplication,
        aStm: CoreAdvancedStatement,
        aliasConfig: AliasConfig,
        userConfig: UserConfig,
        aHmParam: Map<String, String>,
        hmParentData: MutableMap<String, Int>,
        id: Int?
    ) {

        super.init(application, aStm, aliasConfig, userConfig, aHmParam, hmParentData, id)

//----------------------------------------------------------------------------------------------------------------------

        modelTableName = "OFFICE_report"

//----------------------------------------------------------------------------------------------------------------------

        columnId = ColumnInt(modelTableName, "id")

//----------------------------------------------------------------------------------------------------------------------

        val columnUserID = ColumnInt("SYSTEM_users", "id")
        columnReportUser = ColumnInt(modelTableName, "user_id", columnUserID, 0 /*userConfig.getUserID()*/)

        val columnUserName = ColumnString("SYSTEM_users", "full_name", "По пользователю", STRING_COLUMN_WIDTH).apply {
            selectorAlias = "system_user_people"
            addSelectorColumn(columnReportUser, columnUserID)
            addSelectorColumn(this)
        }

        columnSumOnly = ColumnBoolean(modelTableName, "sum_only", "Выводить только суммы", false)

//----------------------------------------------------------------------------------------------------------------------

        alFormHiddenColumn += columnId
        alFormHiddenColumn += columnReportUser

        alFormColumn += columnUserName
        alFormColumn += columnSumOnly
    }
}
